addons/website_profile/static/src/components/profile_dialog/profile_dialog.js

```diff
@@ -0,0 +1,149 @@
+import { onMounted, onWillStart, reactive, useRef, useState } from "@odoo/owl";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { localization } from "@web/core/l10n/localization";
+import { _t } from "@web/core/l10n/translation";
+import { rpc } from "@web/core/network/rpc";
+import { user } from "@web/core/user";
+import { useAutofocus, useService } from "@web/core/utils/hooks";
+import { isHtmlEmpty } from "@web/core/utils/html";
+import { isEmail, uuid } from "@web/core/utils/strings";
+import { setCursorEnd } from "@web_editor/js/editor/odoo-editor/src/utils/utils";
+
+export class ProfileDialog extends ConfirmationDialog {
+    static template = "website_profile.ProfileDialog";
+    static components = {
+        ...ConfirmationDialog.components,
+        Wysiwyg,
+    };
+    static props = {
+        ...ConfirmationDialog.props,
+        focusWebsiteDescription: {
+            type: Boolean,
+            optional: true,
+        },
+        userId: { type: Number },
+    };
+    static defaultProps = {
+        ...ConfirmationDialog.defaultProps,
+        cancel: () => {},
+        cancelLabel: _t("Discard"),
+        confirm: () => true,
+        confirmLabel: _t("Update"),
+        dismiss: () => {},
+        focusWebsiteDescription: false,
+        title: _t("Edit Profile"),
+    };
+
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+        const componentUUID = uuid();
+        this.getUniqueDomID = (suffix) => `profile-dialog-${componentUUID}-${suffix}`;
+        this.upload = useRef("upload");
+        this.profileImg = useRef("profileImg");
+        this.profileImgData = null;
+        this.state = useState({ hasError: false, emailHasError: false, nameHasError: false });
+        useAutofocus({ refName: "name" });
+        const websiteDescriptionClass = this.getUniqueDomID("websiteDescription");
+        onWillStart(async () => {
+            const fields = [
+                "city",
+                "country_id",
+                "email",
+                "name",
+                "website",
+                "website_description",
+                "website_published",
+            ];
+            const [users, countries] = await Promise.all([
+                this.orm.read("res.users", [this.props.userId], fields),
+                this.orm.searchRead("res.country", [], ["id", "name"]),
+            ]);
+            this.user = reactive(users[0], () => this.validate());
+            this.user.country_id = this.user.country_id && this.user.country_id[0]; // keep only id
+            this.countries = countries;
+            const isInternalUser = await user.hasGroup("base.group_user");
+            this.descriptionConfig = {
+                allowFile: isInternalUser,
+                allowImage: isInternalUser,
+                classList: ["form-control", websiteDescriptionClass],
+                content: this.user.website_description,
```

Comment:
@abd-msyukyu-odoo 

Kind of the same question as on Stéphane's PR: do we need to worry about any kind of editor migration here?

Reply:
But any embedded component can be updated right?

For example, if we decide to update the /toc embedded components, what happens if the migration is never run? Does it stop working on "older articles" at some point? Not sure what you mean by "if you don't use it you can ignore it".

STD PR here: https://github.com/odoo/odoo/pull/212175#discussion_r2128282772